ksa_compliance/standard_doctypes/sales_invoice.py

```diff
@@ -156,6 +156,8 @@ def validate_sales_invoice(self: SalesInvoice | POSInvoice, method) -> None:
                 raise_exception=True,
             )
             valid = False
+        else:
```

Comment:
You should not empty the advances
even something wrong if you through exception